### ISSUE TITLE
Set pre-defined connection weights

### ISSUE THREAD

**Issue by jgosmann** [nengo/nengo#767]
It would be nice if one could easily set the connection weight matrix of a connection (e.g., to all zero or small random weights) for some learning stuff.

Maybe this is already possible and I just don't know it because I never done it. But right now the easiest approach seems to me to write a custom solver returning that connection weight matrix.

**Comment by studywolf** [comment]
yeah that's what I did a little while ago when I wanted to do this, it
would be nice to have something in there that just set everything to zero
or random numbers!

**Comment by Seanny123** [comment]
@xchoo told me that you're supposed to make a neuron to neuron connection and then define a transform which will be the connection weights? Did you guys try that?

**Comment by studywolf** [comment]
ah it's a bit trickier because it's for the decoder learning connections, when you're connecting two populations together NEF styles rather than their neurons directly, and so you can't provide a connection weight matrix when creating it

**Comment by hunse** [comment]
If you want to learn decoders from an arbitrary weight matrix, you connect neurons to ensemble.

```python
a = nengo.Ensemble(100, 1)
b = nengo.Ensemble(100, 1)
nengo.Connection(a.neurons, b, transform=np.random.normal(size=(100, 100)))
```

**Comment by tcstewar** [comment]
Just to check, should that example be ```transform=np.random.normal(size=(1, 100))```, @hunse ?

**Comment by jgosmann** [comment]
Does this also work with `np.zeros` instead of `np.random`?

**Comment by hunse** [comment]
Oh yeah, sorry, of course, it should be (1, 100):

```python
a = nengo.Ensemble(100, 1)
b = nengo.Ensemble(100, 1)
nengo.Connection(a.neurons, b, transform=np.random.normal(size=(1, 100)))
```
and it will work with any `ndarray` (or array_like, for that matter).

**Comment by s72sue** [comment]
Is it also possible to learn weights from an arbitrary initial weight matrix when using a weight solver?
weight solvers seem to work only for connections from ensembles (and not neurons).

**Comment by hunse** [comment]
> Is it also possible to learn weights from an arbitrary initial weight matrix when using a weight solver?

The way I see things, you're either learning from random initial weights, in which case your connection has to be from neurons, or you're learning from a "solved" starting point (the output of a decoder or weight solver, not arbitrary), in which case your connection has to be from an ensemble. Hopefully all these cases work, because I'm not sure if they're all properly covered by the unit tests (though I think some of my recent tests helped in this regard).

**Comment by s72sue** [comment]
So for using random initial decoders (where weights is 100x1):
conn = nengo.Connection(input.neurons, answer, transform=weights)

and for using random initial weights (where weights is 100x100):
conn = nengo.Connection(input.neurons, answer.neurons, transform=weights)

is that right?

**Comment by hunse** [comment]
I think `weights` should be 1x100 in your first example, but other than that it looks good.

**Comment by s72sue** [comment]
you are right, it should be 1x100

**Comment by tbekolay** [comment]
Documentation on how to do this is now included as of #1244.
